src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -594,6 +597,17 @@ class BookingService(
         "$.departureDate" hasValidationError "beforeBookingArrivalDate"
       }
 
+      val application = when (assessmentId) {
+        null -> null
+        else -> {
+          val result = assessmentRepository.findByIdOrNull(assessmentId)
+          if (result == null) {
+            "$.assessmentId" hasValidationError "doesNotExist"
```

Comment:
It looks to set an error message on the object that's then validated in the next block which makes sense but I'm curious as to how this line works.

Reply:
The `validated` method creates a [`ValidatedScope`](https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/ValidationErrors.kt), which holds a `MutableMap<String, String>` for validation errors.

The `hasValidationError` function is an [infix function](https://kotlinlang.org/docs/functions.html#infix-notation) which in this case puts `"$.assessmentId" -> "doesNotExist"` into the error map.

Reply:
I didn't know hat an infix function was! So the long hand is `"$.assessmentId".hasValidationError("doesNotExist")`. Thanks :)